### PR TITLE
Multi tenant

### LINKED ISSUE #54: multitenancy support based on an instance ID
**- Do you want to request a *feature* or report a *bug*?**
feature

**- What is the current behavior?**
Currently, a single instance of gocommerce is for a single site. 

**- What is the expected behavior?**
We should have allow for a single instance of gocommerce to support multiple different sites base on the `aud` in the JWT token. 

I am not sure how this will work with anonymous checkouts, but it should be explored.

### PR DESCRIPTION
**Pull request by brycekahle**
Fixes #54

**- Summary**

Add multi-tenant support. Changes config to environment variables only.

**- Test plan**

Existing tests pass. A few tests/assertions added.

**- Description for the changelog**

Add multi-tenant support.

### CODE REVIEW

File: api/instance.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	return sendJSON(w, http.StatusOK, i)
+}
+
+func (a *API) DeleteInstance(w http.ResponseWriter, r *http.Request) error {
+	i := gcontext.GetInstance(r.Context())
+	if err := models.DeleteInstance(a.db, i); err != nil {
+		return internalServerError("Database error deleting instance").WithInternalError(err)
+	}
+
+	// TODO do we delete everything associated with an instance too?
```
Comment (rybit) [inline api/instance.go]: file a ticket

File: cmd/multi_cmd.go [hunk 2]
```diff
@@ -14,25 +14,37 @@ import (
...
-	db, err := models.Connect(config)
+func multi(cmd *cobra.Command, args []string) {
+	globalConfig, err := conf.LoadGlobal(configFile)
+	if err != nil {
+		logrus.Fatalf("Failed to load configuration: %+v", err)
+	}
+	if globalConfig.OperatorToken == "" {
+		logrus.Fatal("Operator token secret is required")
+	}
+	if globalConfig.DB.Namespace != "" {
```
Comment (rybit) [inline cmd/multi_cmd.go]: that isn't required right? The namespace corresponds to the name of the tables, which can be empty. We can use a single set of tables regardless of tenancy

Comment (brycekahle) [inline cmd/multi_cmd.go]: Did you misread something? This is the code:
```
if globalConfig.DB.Namespace != "" {
	models.Namespace = globalConfig.DB.Namespace
}
```

File: api/instance.go (hunk 1 above)
Comment (calavera) [reply api/instance.go]: Let's do hard remove for now. If we want to change it to soft delete later, we can do that all at once.

**Comment by calavera** [comment]
LGTM

File: api/instance.go (hunk 1 above)
Comment (calavera) [reply api/instance.go]: https://github.com/netlify/gocommerce/issues/85

**Pull request merged**